modules/nw-ovn-ipsec-north-south-enable.adoc

```diff
@@ -0,0 +1,138 @@
+
+. Provide the following certificate files to add to the Network Security Services (NSS) database on each host. These files are imported as part of the Butane configuration in subsequent steps.
++
+--
+* `left_server.p12`: The certificate bundle for the IPsec endpoints
+* `ca.pem`: The certificate authority that you signed your certificates with
+--
+
+. Create a machine config to apply the IPsec configuration to your cluster:
+
+.. To add the IPsec configuration, create Butane config files for control plane and worker nodes with the following contents:
++
+[source,terminal,subs="attributes+"]
+----
+$ for role in master worker; do
+  cat >> "99-ipsec-$\{role}-endpoint-config.bu" <<-EOF
+  variant: openshift
+  version: {product-version}.0
+  metadata:
+    name: 99-$\{role}-import-certs-enable-svc-os-ext
+    labels:
+      machineconfiguration.openshift.io/role: $role
+  openshift:
+    extensions:
+      - ipsec
+  systemd:
+    units:
+    - name: ipsec-import.service
+      enabled: true
+      contents: |
+        [Unit]
+        Description=Import external certs into ipsec NSS
+        Before=ipsec.service
+      
+        [Service]
+        Type=oneshot
+        ExecStart=/usr/local/bin/ipsec-addcert.sh
+        RemainAfterExit=false
+        StandardOutput=journal
+
+        [Install]
+        WantedBy=multi-user.target
+    - name: ipsecenabler.service
+      enabled: true
+      contents: |
+        [Service]
+        Type=oneshot
+        ExecStart=systemctl enable --now ipsec.service
+
+        [Install]
+        WantedBy=multi-user.target
+  storage:
+    files:
+    - path: /etc/pki/certs/ca.pem
+      mode: 0400
+      overwrite: true
+      contents:
+        local: ca.pem
+    - path: /etc/pki/certs/left_server.p12
+      mode: 0400
+      overwrite: true
+      contents:
+        local: left_server.p12
+    - path: /usr/local/bin/ipsec-addcert.sh
+      mode: 0740
+      overwrite: true
+      contents:
+        inline: |
+          #!/bin/bash -e
+          echo "importing cert to NSS"
+          certutil -A -n "CA" -t "CT,C,C" -d /var/lib/ipsec/nss/ -i /etc/pki/certs/ca.pem
+          pk12util -W "" -i /etc/pki/certs/left_server.p12 -d /var/lib/ipsec/nss/
+          certutil -M -n "left_server" -t "u,u,u" -d /var/lib/ipsec/nss/
+EOF
+done
+----
+
+.. To transform the Butane files that you created in the previous step to a machine config, enter the following command:
++
+[source,terminal]
+----
+$ for role in master worker; do
+  butane 99-ipsec-${role}-endpoint-config.bu -o ./99-ipsec-$role-endpoint-config.yaml
+done
+----
+
+. To apply the machine configs to your cluster, enter the following command:
```

Comment:
In the previous step, it sounds like we created a single machine config YAML; this step is plural. Do we need to repeat the previous action based on local conditions? Or do the Butane files we created above get made into a single YAML that contains multiple machine configs? The singular/plural in these steps is confusing.

Reply:
@ShaunaDiaz, the for loop creates two machine configs, because reasons.